src/bun.js/bindings/JSBuffer.cpp

```diff
@@ -993,120 +1006,111 @@ static inline JSC::EncodedJSValue jsBufferPrototypeFunction_fillBody(JSC::JSGlob
     }
 
-    {
-        EnsureStillAliveScope value_ = callFrame->argument(0);
-
-        size_t length = castedThis->byteLength();
-        size_t start = 0;
-        size_t end = length;
-        WebCore::BufferEncodingType encoding = WebCore::BufferEncodingType::utf8;
-
-        JSValue encodingValue = jsUndefined();
-        JSValue offsetValue = jsUndefined();
-        JSValue lengthValue = jsUndefined();
-
-        switch (callFrame->argumentCount()) {
-        case 4:
-            encodingValue = callFrame->uncheckedArgument(3);
-            FALLTHROUGH;
-        case 3:
-            lengthValue = callFrame->uncheckedArgument(2);
-            FALLTHROUGH;
-        case 2:
-            offsetValue = callFrame->uncheckedArgument(1);
-            FALLTHROUGH;
-        default:
-            break;
-        }
+    if (!lengthValue.isUndefined()) {
+        end = parseIndex(lexicalGlobalObject, scope, lengthValue);
+        RETURN_IF_EXCEPTION(scope, JSC::JSValue::encode(jsUndefined()));
+    }
 
-        if (offsetValue.isUndefined() || offsetValue.isString()) {
-            encodingValue = offsetValue;
-            offsetValue = jsUndefined();
-        } else if (lengthValue.isString()) {
-            encodingValue = lengthValue;
-            lengthValue = jsUndefined();
-        }
+    if (start >= end) {
+        RELEASE_AND_RETURN(scope, JSValue::encode(castedThis));
+    }
 
-        if (encodingValue.isString()) {
-            std::optional<BufferEncodingType> encoded = parseEnumeration<BufferEncodingType>(*lexicalGlobalObject, encodingValue);
-            if (!encoded) {
-                throwTypeError(lexicalGlobalObject, scope, "Invalid encoding"_s);
-                return JSC::JSValue::encode(jsUndefined());
-            }
+    if (UNLIKELY(end > limit)) {
+        throwRangeError(lexicalGlobalObject, scope, "end out of range"_s);
+        return JSC::JSValue::encode(jsUndefined());
+    }
 
-            encoding = encoded.value();
-        }
+    if (value.isString()) {
+        auto startPtr = castedThis->typedVector() + start;
+        auto str_ = value.toWTFString(lexicalGlobalObject);
+        ZigString str = Zig::toZigString(str_);
 
-        if (!offsetValue.isUndefined()) {
-            if (auto offset = offsetValue.tryGetAsUint32Index()) {
-                start = offset.value();
-            } else {
-                throwVMError(lexicalGlobalObject, scope, createRangeError(lexicalGlobalObject, "start out of range"_s));
-                return JSC::JSValue::encode(jsUndefined());
-            }
+        if (str.len == 0) {
+            memset(startPtr, 0, end - start);
+        } else if (UNLIKELY(!Bun__Buffer_fill(&str, startPtr, end - start, encoding))) {
+            throwTypeError(lexicalGlobalObject, scope, "Failed to decode value"_s);
+            return JSC::JSValue::encode(jsUndefined());
         }
+    } else if (auto* view = JSC::jsDynamicCast<JSC::JSUint8Array*>(value)) {
```

Comment:
we could technically support JSArrayBufferView here if we wanted, it wouldn't cost anything extra

Reply:
I was wondering about that &minus; might as well do it :)

Reply:
Done in https://github.com/oven-sh/bun/pull/2341/commits/e0c5c4aa3a60d39e214b8103ca998744221130fc